### ISSUE TITLE
Libki opens, but displays the taskbar

### ISSUE THREAD

**Issue by TypicalLibrarian** [Libki/libki-client#12]
Lately, when libki boots, the taskbar is visible. Patrons are able to just ignore libki and open whatever app they want. Our clients are running windows 8.1 Enterprise (and two run windows 7).

**Comment by kylemhall** [comment]
It sounds like some software on your system is interfering in some way.

Try this version of the client:
https://github.com/Libki/libki-client/releases/download/Roberto/Wizard__Libki_Client_2.0.2.0_Installer.exe

In theory it will kill the windows explorer when the client launches, and
restart the explorer when a patron logs on. It works for me but hasn't been
tested extensively.

Kyle

<https://secure2.convio.net/cffh/site/Donation2?df_id=1395&FR_ID=4715&PROXY_ID=2706639&PROXY_TYPE=20&1395.donation=form1&s_src=CHORUS&s_subsrc=CHAADOEB>

http://www.kylehall.info
ByWater Solutions ( http://bywatersolutions.com )
Meadville Public Library ( http://www.meadvillelibrary.org )
Crawford County Federated Library System ( http://www.ccfls.org )

**Comment by heyitsjoewevers** [comment]
This is a problem I had with windows 10 as well. The solution I put in place is to run a script that disables user input for 30 seconds, then launches libki. This is a fairly hackey way of doing things, but was the only solution that worked 100% of the time. Different Machines have the task bar pop up more often. In general the better specs on the computer, the less likely it was for this error to occur.

**Comment by kylemhall** [comment]
That's a great idea! Can you share your script with us?

Kyle

<https://secure2.convio.net/cffh/site/Donation2?df_id=1395&FR_ID=4715&PROXY_ID=2706639&PROXY_TYPE=20&1395.donation=form1&s_src=CHORUS&s_subsrc=CHAADOEB>

http://www.kylehall.info
ByWater Solutions ( http://bywatersolutions.com )
Meadville Public Library ( http://www.meadvillelibrary.org )
Crawford County Federated Library System ( http://www.ccfls.org )

**Comment by heyitsjoewevers** [comment]
#include "C:\Users\STAFF\Desktop\MouseOnEvent.au3"

_BlockMouseClicksInput(0)

ToolTip('MouseClicks are disabled')
Sleep(50000)

_BlockMouseClicksInput(1)

ToolTip('MouseClicks are enabled')
Sleep(3000)

Func _BlockMouseClicksInput($iOpt = 0)
	If $iOpt = 0 Then
		_MouseSetOnEvent($MOUSE_PRIMARYUP_EVENT, "__Dummy", 0, 1)
		_MouseSetOnEvent($MOUSE_PRIMARYDOWN_EVENT, "__Dummy", 0, 1)
		_MouseSetOnEvent($MOUSE_SECONDARYUP_EVENT, "__Dummy", 0, 1)
		_MouseSetOnEvent($MOUSE_SECONDARYDOWN_EVENT, "__Dummy", 0, 1)
	Else
		_MouseSetOnEvent($MOUSE_PRIMARYUP_EVENT)
		_MouseSetOnEvent($MOUSE_PRIMARYDOWN_EVENT)
		_MouseSetOnEvent($MOUSE_SECONDARYUP_EVENT)
		_MouseSetOnEvent($MOUSE_SECONDARYDOWN_EVENT)
	EndIf
EndFunc

Save this as blank.au3 

It is an autoITv3 script. It basically disables the mouse for 50 seconds. You will need to have auto IT installed to run it. I believe I used a reg key to place this script before the userlogin script. I'm sorry that this might not be very helpful, It has been about a year since I used either AutoIT or Libki and I'm a bit fuzzy on the details.

**Comment by loidor** [comment]
I'd suggest to also disable the win key for some seconds as well. Don't know if autoit can do it, but I'm pretty sure autohotkey should be able to pull off both.

**Comment by kylemhall** [comment]
Thanks! Since we are already using autohotkey in Libki, I used your script
as inspiration and was able to replicate your idea!

If anyone is interested in testing, just let me know!

Kyle

<https://secure2.convio.net/cffh/site/Donation2?df_id=1395&FR_ID=4715&PROXY_ID=2706639&PROXY_TYPE=20&1395.donation=form1&s_src=CHORUS&s_subsrc=CHAADOEB>

http://www.kylehall.info
ByWater Solutions ( http://bywatersolutions.com )
Meadville Public Library ( http://www.meadvillelibrary.org )
Crawford County Federated Library System ( http://www.ccfls.org )

**Comment by kylemhall** [comment]
If memory serves me, I think I tried to disable the Windows key and failed.
If you want to give it a shot please do!

Kyle

<https://secure2.convio.net/cffh/site/Donation2?df_id=1395&FR_ID=4715&PROXY_ID=2706639&PROXY_TYPE=20&1395.donation=form1&s_src=CHORUS&s_subsrc=CHAADOEB>

http://www.kylehall.info
ByWater Solutions ( http://bywatersolutions.com )
Meadville Public Library ( http://www.meadvillelibrary.org )
Crawford County Federated Library System ( http://www.ccfls.org )

**Comment by loidor** [comment]
I don't have win access right now, but shouldn't it just be 
```
LWin::return
RWin::return
```
Or is this one of those "you must run this as admin" thingies? That might make things more troublesome.

**Comment by heyitsjoewevers** [comment]
Great, glad I could help a little bit!

**Comment by kylemhall** [comment]
Yes, I think that may be the problem. UAC has made it more difficult to
override a lot of Windows behavior without additional modification the
client itself cannot do.

Kyle

<https://secure2.convio.net/cffh/site/Donation2?df_id=1395&FR_ID=4715&PROXY_ID=2706639&PROXY_TYPE=20&1395.donation=form1&s_src=CHORUS&s_subsrc=CHAADOEB>

http://www.kylehall.info
ByWater Solutions ( http://bywatersolutions.com )
Meadville Public Library ( http://www.meadvillelibrary.org )
Crawford County Federated Library System ( http://www.ccfls.org )
